test/uk/gov/hmrc/entrydeclarationstore/services/SubmissionReplayServiceSpec.scala

```diff
@@ -0,0 +1,183 @@
+import uk.gov.hmrc.entrydeclarationstore.reporting.{MockReportSender, SubmissionSentToEIS}
+import uk.gov.hmrc.entrydeclarationstore.repositories.{MetadataLookupError, MockEntryDeclarationRepo}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.test.UnitSpec
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class SubmissionReplayServiceSpec
+    extends UnitSpec
+    with MockEntryDeclarationRepo
+    with MockEisConnector
+    with MockReportSender
+    with ScalaFutures {
+
+  val service = new SubmissionReplayService(mockEntryDeclarationRepo, mockEisConnector, mockReportSender)
+
+  val eori          = "eori"
+  val correlationId = "correlationId"
+  val now: Instant  = Instant.now
+  val subId1        = "subId1"
+  val subId2        = "subId2"
+
+  implicit val hc: HeaderCarrier = HeaderCarrier()
+
+  def metadata(submissionId: String): EntryDeclarationMetadata =
+    EntryDeclarationMetadata(submissionId, MessageType.IE315, "5", now, None)
+
+  def replayMetadata(submissionId: String): ReplayMetadata = ReplayMetadata(eori, correlationId, metadata(submissionId))
+
+  private def submissionSentToEISReport(submissionId: String, failure: Option[EISSendFailure]) =
+    SubmissionSentToEIS(
+      eori          = eori,
+      correlationId = correlationId,
+      submissionId,
+      MessageType.IE315,
+      failure
+    )
+
+  "SubmissionReplayService" when {
+    "replaying a submission" when {
+      "successful" must {
+        "increment success count" in {
+          val submissionIds = Seq(subId1)
+
+          val report = submissionSentToEISReport(submissionIds.head, None)
+          MockEntryDeclarationRepo.lookupMetadata(submissionIds.head) returns Future.successful(
+            Right(replayMetadata(subId1)))
+          MockEisConnector.submitMetadata(metadata(subId1)) returns Future.successful(None)
+          MockReportSender.sendReport(report) returns Future.successful((): Unit)
+
+          service.replaySubmission(submissionIds).futureValue shouldBe Right(ReplayResult(1, 0))
+        }
+        "increment success counts" in {
+          val submissionIds = Seq(subId1, subId2)
+          submissionIds.foreach { submissionId =>
+            val report = submissionSentToEISReport(submissionId, None)
+            MockEntryDeclarationRepo
+              .lookupMetadata(submissionId)
+              .returns(Future.successful(Right(replayMetadata(submissionId))))
+            MockEisConnector.submitMetadata(metadata(submissionId)) returns Future.successful(None)
+            MockReportSender.sendReport(report) returns Future.successful((): Unit)
+          }
+
+          service.replaySubmission(submissionIds).futureValue shouldBe Right(ReplayResult(2, 0))
+        }
+      }
+
+      "Metadata retrieval unsuccessful" must {
+        "increment failure count" in {
```

Comment:
```suggestion
        "increment failure count" when {
          Seq(MetadataLookupError.DataFormatError, MetadataLookupError.MetadataNotFound) foreach incrementFailureCountFor

          def incrementFailureCountFor(err: MetadataLookupError): Unit = s"it fails with $err" in {
```

or something like that (may need another closing brace) to make sure treat both failure types alike